code/modules/surgery/organs/heart.dm

```diff
@@ -254,3 +254,192 @@
+		deltimer(crystalize_timer_id)
+		crystalize_timer_id = null
+
+
+///Lets you stop the process with enough brute damage
+/obj/item/organ/heart/ethereal/proc/on_take_damage(datum/source, damage, damagetype, def_zone)
+	SIGNAL_HANDLER
+	if(damagetype != BRUTE)
+		return
+
+	crystalization_process_damage += damage
+
+	if(crystalization_process_damage < BRUTE_DAMAGE_REQUIRED_TO_STOP_CRYSTALIZATION)
+		return
+
+	var/mob/living/carbon/human/ethereal = source
+
+	to_chat(ethereal, "<span class='userwarning'>The crystals on your body have completely broken</span>")
+	ethereal.visible_message("<span class='notice'>The crystals on [ethereal] are completely shattered and stopped growing</span>")
+
+	stop_crystalization_process(ethereal)
+
+/obj/structure/ethereal_crystal
+	name = "Ethereal Ressurection Crystal"
+	desc = "It seems to contain the corpse of an Ethereal mending its wounds."
+	icon = 'icons/obj/ethereal_crystal.dmi'
+	icon_state = "ethereal_crystal"
+	damage_deflection = 0
+	max_integrity = 100
+	resistance_flags = FIRE_PROOF
+	density = TRUE
+	anchored = TRUE
+	///The organ this crystal belongs to
+	var/obj/item/organ/heart/ethereal/ethereal_heart
+	///Timer for the healing process. Stops if destroyed.
+	var/crystal_heal_timer
+
+/obj/structure/ethereal_crystal/Initialize(mapload, obj/item/organ/heart/ethereal/ethereal_heart)
+	. = ..()
+	src.ethereal_heart = ethereal_heart
+	ethereal_heart.owner.visible_message("<span class='notice'>The crystals fully encase [ethereal_heart.owner]</span>")
+	to_chat(ethereal_heart.owner, "<span class='notice'>You are encased in a huge crystal.</span>")
+	playsound(get_turf(src), 'sound/effects/ethereal_crystalization.ogg', 50)
+	ethereal_heart.owner.forceMove(src) //put that ethereal in
+	add_atom_colour(ethereal_heart.ethereal_color, FIXED_COLOUR_PRIORITY)
+	crystal_heal_timer = addtimer(CALLBACK(src, .proc/heal_ethereal), CRYSTALIZE_HEAL_TIME, TIMER_STOPPABLE)
+	set_light(4, 10, ethereal_heart.ethereal_color)
+	update_icon()
+
+/obj/structure/ethereal_crystal/obj_destruction(damage_flag)
+	playsound(get_turf(ethereal_heart.owner), 'sound/effects/ethereal_revive_fail.ogg', 100)
+	return ..()
+
+
+/obj/structure/ethereal_crystal/Destroy()
+	if(!ethereal_heart)
+		return ..()
+	ethereal_heart.current_crystal = null
+	COOLDOWN_START(ethereal_heart, crystalize_cooldown, CRYSTALIZE_COOLDOWN_LENGTH)
+	ethereal_heart.owner.forceMove(get_turf(src))
+	REMOVE_TRAIT(ethereal_heart.owner, TRAIT_CORPSELOCKED, SPECIES_TRAIT)
+	deltimer(crystal_heal_timer)
+	visible_message("<span class='notice'>The crystals shatters, causing [ethereal_heart.owner] to fall out</span>")
+	return ..()
+
+/obj/structure/ethereal_crystal/update_overlays()
+	. = ..()
+	var/mutable_appearance/shine = mutable_appearance(icon, icon_state = "[icon_state]_shine")
+	shine.appearance_flags = RESET_COLOR //No color on this, just pure white
+	. += shine
+
+/obj/structure/ethereal_crystal/proc/heal_ethereal()
+	ethereal_heart.owner.revive(TRUE, FALSE)
```

Comment:
I'd recommend making this set your race to being an ethereal, because the idea of using an crystaline heart to reforge a dead guy into a new member of an alien race sounds fucking badass

also, uh, what happens if a xenomorph (queen) revives using an ethereal heart? do they still get the brain trauma?

Reply:
yes